index.js

```diff
@@ -125,5 +136,11 @@ const getColor = (percentile) => {
 sense.Leds.lowLight = true
 sense.Leds.clear()
 sense.Leds.flipH()
-run()
 
+run();
+
+// If we're event driven, we only ever do something if an event drops on us
+if (eventDriven) {
+  // Initiate
+  frontInstance.registerEvents({ port: 80 }, run);
```

Comment:
This line means that every time the Front SDK sees an event, it will call the run method. So the action taken is literally the same as the polling method, but that it doesn't need timers to operate. If no events occur on Front, run isn't called.
The end of the run method sets a timer if the event model isn't being used, or simply exits to await another event.